src/main/java/org/jabref/gui/entryeditor/DeprecatedFieldsTab.java

```diff
@@ -53,7 +54,14 @@ public DeprecatedFieldsTab(BibDatabaseContext databaseContext,
     protected Set<Field> determineFieldsToShow(BibEntry entry) {
         Optional<BibEntryType> entryType = entryTypesManager.enrich(entry.getType(), databaseContext.getMode());
         if (entryType.isPresent()) {
-            return entryType.get().getDeprecatedFields();
+            Set<Field> validDeprecatedFields = new HashSet<>();
```

Comment:
You can make this a one liner with Streams:

`            return entryType.get().getDeprecatedFields().stream().filter(field->! entry.getField(field).isEmpty()).collect(Collectors.toSet());
`